src/entries/popup/hooks/useRestoreNavigation.ts

```diff
@@ -74,7 +74,11 @@ const RESTORE_NAV_MAP: Record<string, string[]> = {
   [ROUTES.SETTINGS__NETWORKS]: [ROUTES.SETTINGS, ROUTES.SETTINGS__NETWORKS],
   [ROUTES.SETTINGS__NETWORKS__CUSTOM_RPC]: [
     ROUTES.SETTINGS,
-    ROUTES.SETTINGS__NETWORKS__CUSTOM_RPC,
+    ROUTES.SETTINGS__NETWORKS,
+  ],
+  [ROUTES.SETTINGS__NETWORKS__CUSTOM_RPC__DETAILS]: [
+    ROUTES.SETTINGS,
+    ROUTES.SETTINGS__NETWORKS,
```

Comment:
Noticed that on this page I was getting dropped all the way back to home, so I'm at least getting the user a little bit closer to where they left off.

Reply:
what if we change the url to pass the chain via query string? That should be a very small change and would fix this issue completely.

Reply:
https://recordit.co/tamNg9gQ4I

Alright, went with persisting location.state but this worked in fixing the paths for both routes